weldx/asdf/tags/weldx/core/data_array.py

```diff
@@ -69,12 +69,8 @@ def from_tree(cls, tree, ctx):
         """
         data = tree["data"].data
         dims = tree["data"].dimensions
-        coords = {}
-        for coordinate in tree["coordinates"]:
-            coords[coordinate.name] = (coordinate.dimensions, coordinate.data)
-
-        obj = DataArray(data=data, coords=coords, dims=dims)
-
-        obj.attrs = tree["attributes"]
+        coords = {c.name: (c.dimensions, c.data, c.attrs) for c in tree["coordinates"]}
+        attrs = tree["attributes"]
 
-        return obj
+        da = DataArray(data=data, coords=coords, dims=dims, attrs=attrs)
+        return da
```

Comment:
think we can tighten the code a bit more:

~~~
coords = {c.name: (c.dimensions, c.data, c.attrs) for c in tree["coordinates"]}

return DataArray(
    data=tree["data"].data, 
    coords=coords, 
    dims=tree["data"].dimensions, 
    attrs=tree["attributes"]
)
~~~